src/components/ToDoItem.tsx

```diff
@@ -0,0 +1,138 @@
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { Dispatch } from './ToDoContext';
+
+type Props = {
+  todo: Todo;
+  key: number;
```

Comment:
The key prop should not be passed as a part of the component’s props. React automatically handles the key prop when rendering lists of components.